src/ispc/kernels/lanczos3.ispc

```diff
@@ -28,22 +48,27 @@ static inline float lanczos3_filter(float t)
         return 0.0f;
 }
 
-struct Image {
-    uniform const uint8* data;
-    uniform int<2> size;
-};
-
 static inline float frac(float f) {
     float absf = abs(f);
     return absf - floor(absf);
 }
 
-static inline float byte_to_float(uint b) {
-    //return floatbits(0x3f800000 | (b << (23 - 8))) - 1.0;
-    return (float)b;
+static inline float byte_to_float(uint8 b) {
+    const float inv_255 = rcp(255.0);
+
+    // floatbits(0x3f800000 | (b << (23 - 8))) - 1.0;
+    return (float)b * inv_255;
+}
+
+static inline uint8 float_to_byte(float d, bool gamma) {
+    if (gamma) {
+        d = pow(d, 1.0f / GAMMA);
+    }
```

Comment:
This should be a `uniform bool`, though ideally we hoist this branch out all the way.